priiloader/source/main.cpp

```diff
@@ -2832,7 +2861,58 @@ int main(int argc, char **argv)
 	StateFlags flags = GetStateFlags();
 	gprintf("Bootstate %u detected. DiscState %u ,ReturnTo %u & Flags %u & checksum %u (gcflag : 0x%08X)", flags.type, flags.discstate, flags.returnto, flags.flags, flags.checksum, GcShutdownFlag);
 	s8 magicWord = CheckMagicWords();
-	
+
+	if (CheckvWii())
+	{
+		ImportWiiUConfig();
+
+		// bit of a hack but BC-NAND clears the boot state, so let's try to guess it
+		if (!IsVWiiInitialBoot())
+			Bootstate = TYPE_RETURN;
+		else
+			Bootstate = TYPE_UNKNOWN;
+
+		flags.type = Bootstate;
+		SetBootState(flags.type, flags.flags, flags.returnto, flags.discstate);
+
+		// Set video standard type
+		CONF_Init();
+		s32 video = CONF_GetVideo();
+		*(vu32*)0x800000CC = video;
+
+		// Configure proper vWii video mode
+		if (IsVWiiInitialBoot() || VIDEO_GetVideoScanMode() == VI_NON_INTERLACE)

```

Comment:
shouldn't the video init thing only be done if priiloader has init video?

Reply:
Nah, `__VIInit` is now patched out of the SM so priiloader needs to do it instead, even if priiloader doesn't need video.